homeassistant/components/config/automation.py

```diff
@@ -80,3 +89,21 @@ def _write_value(self, hass, data, config_key, new_value):
         updated_value.update(cur_value)
         updated_value.update(new_value)
         data[index] = updated_value
+
+
+@websocket_api.websocket_command(
+    {vol.Required("type"): "automation/trace", vol.Optional("automation_id"): str}
+)
+@websocket_api.async_response
+async def websocket_automation_trace(hass, connection, msg):
+    """Get automation traces."""
+    automation_id = msg.get("automation_id")
+
+    if not automation_id:
+        automation_traces = get_debug_traces(hass)
```

Comment:
This will be a lot of data. I wonder if we should have a "overview" output that gives just the names and timestamps of the last runs that a user can drill into.

For now I would just leave support for leaving `automation_id` out .

Reply:
"overview" output added in #47557